runtime/src/accounts_db.rs

```diff
@@ -9950,6 +9995,33 @@ pub mod tests {
         }
     }
 
+    #[test]
+    fn test_combine_multiple_slots_into_one_at_startup() {
+        solana_logger::setup();
+        let (db, slot1) = create_db_with_storages_and_index(false, 2, None);
+        let slot2 = slot1 + 1;
+
+        let initial_accounts = get_all_accounts(&db, slot1..(slot2 + 1));
+
+        let tf = TempDir::new().unwrap();
+        let stores = db
+            .storage
+            .all_slots()
+            .into_iter()
+            .map(|slot| {
+                let storage = db.storage.get_slot_storage_entry(slot).unwrap();
+                (storage.append_vec_id(), storage)
+            })
+            .collect::<HashMap<_, _>>();
+        let new_storage =
+            AccountsDb::combine_multiple_slots_into_one_at_startup(tf.path(), 1000, slot1, &stores);
```

Comment:
I'm not sure, I fully understand all the logic code that is used to generate input in this test.
But it seems to me, this creates a single storage per slot, does it not?
If so, would it make sense to also add a test where there are several versions of storage files per slot?  To exercise the combination logic.
At the moment, this combination logic is rather simple and is obviously correct.
But as the test would encode the intent, it might save someone breaking it accidentally in the future.

Reply:
I just needed to create 2 storages. It was just convenient framework code to create storages.